### PR TITLE
Product Variations Mark 2: Networking!

### PR DESCRIPTION
**Pull request by bummytime**
This PR adds the networking logic needed for product variations. As always, the size of this PR reflects a lot of testing JSON 😃 so don't be scared.

Stuff added in this PR:
* `ProductVariationListMapper`
* `ProductVariationMapper`
* `ProductVariation` model
* `ProductVariationAttribute` model (this is different from the `ProductAttribute` therefore a different model is used.
* Variation fetch methods were added to `ProductsRemote`
* Lots of unit tests
* I fixed a few misc doc comments in Networking (Products & Orders)

**Note:** In the name of reuse, `ProductDimensions` and `ProductImage` were reused for variations since the objects are identical to the ones `Product` uses.

Ref: #818 

## Testing
In short, look over the code here and make sure the unit tests pass. I will be bringing everything together in next Mark 3 PR. 

@astralbodies, @mindgraffiti, @ctarda — I would like to land this in the 1.6 build, so if anybody could give me a quick review, I would greatly appreciate it. 😀

### CODE REVIEW

**Comment by bummytime** [comment]
Thank you for the quick review @mindgraffiti !!! 🙇 🥇
